tensorflow_recommenders/layers/factorized_top_k_test.py

```diff
@@ -61,6 +61,20 @@ def test_cases(
 
 class FactorizedTopKTestBase(tf.test.TestCase, parameterized.TestCase):
 
+  def run_save_and_restore_test(self, layer, query, num):
+    for _ in range(num):
+      pre_serialization_results = layer(query)
+
+    path = os.path.join(self.get_temp_dir(), "layer")
+    layer.save(
+      path, options=tf.saved_model.SaveOptions(namespace_whitelist=["Scann"]))
+    restored = tf.keras.models.load_model(path)
+
+    for _ in range(num):
+      post_serialization_results = restored(tf.constant(query))
+
+    self.assertAllEqual(post_serialization_results, pre_serialization_results)
+
```

Comment:
Whether it is possible to return the same result after reloading is an important feature.
This process should be used in many test cases, so I split it.